### PR TITLE
Create a new Target for CustomContentTypes pagebuilder

### LINKED ISSUE #2968: [feature]: Extensibility point (new target) to Pagebuilder setContentTypeConfig
Now we have a good way to add pagebuilder content type with setContentTypeConfig 

https://github.com/magento/pwa-studio/blob/develop/packages/pagebuilder/lib/config.js#L29

https://github.com/magento/pwa-studio/issues/2130
https://github.com/magento/pwa-studio/pull/2131

but setContentTypeConfig is hard to using 

I must use my custom babel plugin to insert my "component" before each <RichContent ... />
 in "@magento/venia-ui/lib/RootComponents/CMS/cms", etc.
it is the wrong way because I need to find every <RichContent> usage

so I think the next good step will be to add (declare) pagebuilder target to this point

API 
intercept.js in custom module
```js
module.exports = targets => {
targets
        .of('@magento/pagebuilder')
        .contentTypes.tap(contentTypes => {
            contentTypes.add({
                componentName: '[some]ContentType',
                importPath: './lib/[some]'
            });
        });
};
```

some.js
```
import React from 'react';
import { setContentTypeConfig } from '@magento/pagebuilder/lib/config';

import Component from './component';
import configAggregator from './configAggregator';

setContentTypeConfig('some', {
    configAggregator,
    component: Component
});
```

**Please let us know what packages this feature is in regards to:**
- [x] pagebuilder
- [x] `venia-concept`
- [x] `venia-ui`
- [ ] `pwa-buildpack`
- [x] `peregrine`
- [x] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

### PR DESCRIPTION
**Pull request by 0m3r**
## Description

It gets the possibility for an extension to add custom [content type renderers](https://github.com/magento/pwa-studio/blob/develop/packages/pagebuilder/lib/config.js#L29). 

https://github.com/magento/pwa-studio/blob/develop/packages/pagebuilder/lib/config.js#L29

## Related Issue

#2130
#2131
#2968

Closes #2968

## Acceptance
@supernova-at
@larsroettig

## Verification Steps

`intercept.js`
```js
    targets
        .of('@magento/pagebuilder')
        .customContentTypes.tap(contentTypes => {
            contentTypes.add({
                contentType: 'XSlider',
                importPath: require.resolve(
                    "../lib/components/XSlider/ContentTypes/XSlider/index.js"
                )
            });
        });
```

`lib/components/XSlider/ContentTypes/configAggregator.js`
```js
export default node => {
    console.log(node);
    return {
        // your config for component
    };
};
```

`lib/components/XSlider/ContentTypes/XSlider/index.js`

```js
import Component from './xslider';
import configAggregator from './configAggregator';

export default {
    name: 'xslider',
    configAggregator,
    component: Component
}
```

### CODE REVIEW

**Comment by larsroettig** [comment]
Hi @0m3r, is this ready to review then I can do it today

**Comment by 0m3r** [comment]
>Hi @0m3r, is this ready to review then I can do it today?

Yes, that is all from me now.
Unfortunately, I'm not qualified to write good tests for this.

**Comment by larsroettig** [comment]
Hi  @0m3r. 
For tests, pls take a look at https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/targets/__tests__/venia-ui-targets.spec.js. Also, you should run `yarn run prettier`. 

@jimbo / @fooman   I would know your opinion on this PR main question should we redefine `slider` by an extension or not from my point of view. We should allow overwriting the PageBuilder element by this extension point. But I don't test if it is possible with this PR.

**Comment by fooman** [comment]
@larsroettig Don't really have much to add to the discussion. From my point of view the more extension points the better. So yes seeing that @0m3r went as far as providing a PR more than confirms that this is sought after capability.

Your earlier comment that extension points should not create a new dependency made a lot of sense to me.

File: packages/pagebuilder/lib/factory.js
Comment (larsroettig) [inline packages/pagebuilder/lib/factory.js:8]: Pls, add test coverage to: 
https://github.com/magento/pwa-studio/blob/1977f38305ff6c0e2b23a9da7beb0b2f69758bed/packages/pagebuilder/lib/__tests__/factory.spec.js

File: packages/pagebuilder/lib/ContentTypes/CustomContentTypeList.js
Comment (larsroettig) [inline packages/pagebuilder/lib/ContentTypes/CustomContentTypeList.js:26]: strategy is not used as parameter here

File: packages/pagebuilder/lib/factory.js
Comment (0m3r) [reply packages/pagebuilder/lib/factory.js:8]: I didn't export this function before, should I do it now for the test?

Comment (larsroettig) [reply packages/pagebuilder/lib/factory.js:8]: sorry this is true :smile_cat: I am fine with it don't make sense to export to get coverage here

**Review by larsroettig — APPROVED** [review]
@0m3r I push to core team need the second option can take some time before they can Review and QA it.

**Review by jimbo — COMMENTED** [review]
This looks like the right approach to me. Defines an array, exposes that array so extensions can add to it, and consumes that array when setting up content types.

Someone else should verify the PB-specific code here, but I think this is on the right track.

**Comment by tjwiebell** [comment]
Some Cypress failures unrelated to this change that I'll create a followup PR to resolve. Coverage decrease was from existing code, so not going to block on that. QA Approved ✅

**Pull request merged**
